internal/core/domain/trade_model.go

```diff
@@ -1,64 +1,299 @@
 package domain
 
 import (
+	"reflect"
+
 	"github.com/google/uuid"
-	pb "github.com/tdex-network/tdex-protobuf/generated/go/operator"
+	pkgswap "github.com/tdex-network/tdex-daemon/pkg/swap"
+	"github.com/tdex-network/tdex-daemon/pkg/transactionutil"
+	"github.com/tdex-network/tdex-daemon/pkg/wallet"
+	pbswap "github.com/tdex-network/tdex-protobuf/generated/go/swap"
+	"google.golang.org/protobuf/proto"
 )
 
 var (
-	// EmptyStatus represent the status of an empty trade
+	// EmptyStatus represent the status of an empty trade.
 	EmptyStatus = Status{
-		Code: pb.SwapStatus_UNDEFINED,
+		Code: Empty,
 	}
 	// ProposalStatus represent the status of a trade presented by some trader to
-	// the daemon and not yet processed
+	// the daemon and not yet processed.
 	ProposalStatus = Status{
-		Code: pb.SwapStatus_REQUEST,
+		Code: Proposal,
 	}
 	// ProposalRejectedStatus represents the status of a trade presented by some
-	// trader to the daemon and rejected for some reason
+	// trader to the daemon and rejected for some reason.
 	ProposalRejectedStatus = Status{
-		Code:   pb.SwapStatus_REQUEST,
+		Code:   Proposal,
 		Failed: true,
 	}
 	// AcceptedStatus represents the status of a trade proposal that has been
 	// accepted by the daemon
 	AcceptedStatus = Status{
-		Code: pb.SwapStatus_ACCEPT,
+		Code: Accepted,
 	}
 	// FailedToCompleteStatus represents the status of a trade that failed to be
-	// be completed for some reason
+	// be completed for some reason.
 	FailedToCompleteStatus = Status{
-		Code:   pb.SwapStatus_ACCEPT,
+		Code:   Accepted,
 		Failed: true,
 	}
 	// CompletedStatus represents the status of a trade that has been completed
-	// and accepted in mempool, waiting for being published on the blockchain
+	// and accepted in mempool, waiting for being published on the blockchain.
 	CompletedStatus = Status{
-		Code: pb.SwapStatus_COMPLETE,
+		Code: Completed,
+	}
+	// FailedToSettleStatus represents the status of a trade that failed to be
+	// be settled for some reason.
+	FailedToSettleStatus = Status{
+		Code:   Completed,
+		Failed: true,
+	}
+	// SettledStatus represents the status of a trade that has been settled,
+	// meaning that has been included into the blockchain.
+	SettledStatus = Status{
+		Code: Settled,
+	}
+)
+
+// AcceptArgs ...
+type AcceptArgs struct {
+	RequestMessage     []byte
+	Transaction        string
+	InputBlindingKeys  map[string][]byte
+	OutputBlindingKeys map[string][]byte
+}
+
+// SwapError is the special error returned by the ISwapParser when serializing
+// a swap message.
+type SwapError struct {
+	Err  error
+	Code int
+}
+
+func (s *SwapError) Error() string {
+	return s.Err.Error()
+}
+
+// ISwapParser defines the required methods to override the default swap
+// message parser, which is grpc-proto.
+type ISwapParser interface {
```

Comment:
I  dont see benefit of grouping these methods as interface:
1. we wont have more implementations of it
2. it doesnt have impact on testing

also i would methods to domain/trade_service.go

Reply:
it does have impact on testing instead. Without abstracting components like the swap parser it wouldn't just be possible to use mocks when testing. This is the reason why we had "real" mocks for trade domain tests before this. Take a look at them before and after this PR. I do think these abstractions bring benefits, actually, in particular when testing.

Also, this is a nice way to be able to select which type of MnemonicStore we want to use. At the moment the default MnemonicStore is the "config" one, which lives inside the Vault domain. They should be decoupled from each other, though, so the plan would be to move that implementation outside the domain. This wouldn't just be possible without such abstractions. But, guys, if you have another solution in mind, please share so we can discuss about that.